src/DotNetty.Transport/Bootstrapping/Bootstrap.cs

```diff
@@ -146,9 +146,9 @@ public Task<IChannel> ConnectAsync(EndPoint remoteAddress, EndPoint localAddress
             {
                 resolvedAddress = await this.resolver.ResolveAsync(remoteAddress);
             }
-            catch (Exception ex)
+            catch (Exception)
             {
-                channel.CloseAsync();
+                await channel.CloseAsync();
```

Comment:
if this throws, it would mask an original error. It's better to have a try-catch around it and log using `Logger`.